invenio_access/models.py

```diff
@@ -134,3 +137,38 @@ class ActionRoles(ActionNeedMixin, db.Model):
     def need(self):
         """Return RoleNeed instance."""
         return RoleNeed(self.role.name)
+
+
+def removed_or_inserted_action(mapper, connection, target):
+    """Remove the action from cache when an item is inserted or deleted."""
+    current_access.delete_action_cache(target.action)
+
+
+def changed_action_action(target, value, oldvalue, initiator):
+    """Remove the action from cache when an action is updated.
+
+    Remove the action from cache when ActionRoles.action or ActionUsers.action
+    is updated.
+    """
+    current_access.delete_action_cache(value)
+    current_access.delete_action_cache(oldvalue)
+
+
+def changed_owner_action(target, value, oldvalue, initiator):
```

Comment:
I was using here ```del``` to remove the unused parameters in order to avoid the QuantifiedCode issues.